src/D2L.Bmx/UpdateHandler.cs

```diff
@@ -0,0 +1,130 @@
+using System.Formats.Tar;
+using System.IO.Compression;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace D2L.Bmx;
+
+internal class UpdateHandler {
+
+	public async Task HandleAsync() {
+		using var httpClient = new HttpClient();
+		var releaseData = await UpdateChecker.GetLatestReleaseDataAsync();
+		var localVersion = Assembly.GetExecutingAssembly().GetName().Version;
+		var latestVersion = new Version( UpdateChecker.GetLatestReleaseVersion( releaseData ) );
+		if( latestVersion <= localVersion ) {
+			Console.WriteLine( $"Already own the latest version {latestVersion}" );
+			return;
+		}
+		string archiveName = GetOSFileName();
+		string downloadUrl = releaseData?.Assets?.FirstOrDefault( a => a.Name == archiveName )?.BrowserDownloadUrl
+			?? string.Empty;
+		if( string.IsNullOrWhiteSpace( downloadUrl ) ) {
+			throw new BmxException( "Failed to get update download url" );
+		}
+
+		string? downloadPath = Path.GetTempFileName();
+
+		string? currentProcessPath = Environment.ProcessPath;
```

Comment:
```suggestion
		string? currentProcessPath = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
```

Looks like `Environment.ProcessPath` isn't what we want and it is placing the old outdated file in the current directory your shell has

Reply:
Which means the old versions are going to be placed in probably the wrong spot and won't get cleaned up

Reply:
`string baseDirectory = AppContext.BaseDirectory;` can get you the folder where bmx is. `Process.GetCurrentProcess().MainModule!.FileName` can get you the executable name

Reply:
I made a branch [here](https://github.com/Brightspace/bmx/commit/86fdf104ecb4513e95079db5df6bb516754c3eee) trying it out

Reply:
Strange? I thought Environment.ProcessPath is fine to grab the current file but we can place the old outdated version in another folder and modify the cleanup for that aswell

Reply:
Here's what you can do to reproduce:
- Build bmx and replace the executable C:/bin
- Open powershell and pick any folder
- type `bmx update` and you will find that the bmx.*.old.bak is in that folder you chose above

Reply:
But yea moving to .bmx on the rename should fix it

Reply:
Ah gotcha thanks for the help